Makefile

```diff
@@ -157,15 +157,21 @@ kind-deploy-controller: generate-operator-yaml install-resources deploy
 .PHONY: deploy-controller
 deploy-controller: kind-deploy-controller
 
+HOSTED ?= none
+
 .PHONY: kind-deploy-controller-dev
-kind-deploy-controller-dev: kind-deploy-controller build-images
+kind-deploy-controller-dev:
+	if [ "$(HOSTED)" != "hosted" ]; then\
+		make kind-deploy-controller build-images;\
+	fi;
 	@echo Pushing image to KinD cluster
 	kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)
 	@echo "Patch deployment image"
 	kubectl patch deployment $(IMG) -n $(KIND_NAMESPACE) -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"$(IMG)\",\"imagePullPolicy\":\"Never\"}]}}}}"
 	kubectl patch deployment $(IMG) -n $(KIND_NAMESPACE) -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"$(IMG)\",\"image\":\"$(REGISTRY)/$(IMG):$(TAG)\"}]}}}}"
 	kubectl rollout status -n $(KIND_NAMESPACE) deployment $(IMG) --timeout=180s
 
+
```

Comment:
```suggestion
```

Reply:
Yes this called in framewok. upstream we don't call Framework so

Reply:
`        echo "::group::make build-images"
        make build-images
        echo "::endgroup::"

        echo "::group::make kind-deploy-controller-dev"
        make kind-deploy-controller-dev
        echo "::endgroup::"
`

line 159  https://github.com/stolostron/governance-policy-framework/blob/0903452ed5de2b5d99eba286f4a16ef1007ed064/.github/workflows/kind.yml#L159

Reply:
Gotcha, thanks! This is a good solution, but I'll look at the flows a bit in the framework and other repos since it seems rather specific to this repo and I'm wondering how the other repos compare.

Also, the blank line here still needs to be removed 🙂

Reply:
Thanks! I just removed it!

Reply:
Furthermore, addon-controller, propagator I should use other way

Reply:
only problem is namespace

Reply:
It should be "cluster2-hosted" KIND_NAMESPACE https://github.com/stolostron/governance-policy-framework/blob/0903452ed5de2b5d99eba286f4a16ef1007ed064/.github/workflows/kind.yml#L144

Reply:
Hmmm. In that case, I think we should swap the dependencies that `CONTROLLER_NAMESPACE` and `KIND_NAMESPACE` have on each other. Could we do this instead at the top of the file?

```make
KIND_NAMESPACE ?= open-cluster-management-agent-addon
CONTROLLER_NAMESPACE ?= $(KIND_NAMESPACE)
```

Reply:
only hosted mode case, normal mode use CONTROLLER_NAMESPACE namesapce

Reply:
Right, but currently the default is set on `CONTROLLER_NAMESPACE`. If we swap it, then the framework can override it. Or are you saying that wouldn't work for the normal mode?

Reply:
In the normal mode, KIND_NAMESPACE=managed so the controller should be deployed in CONTROLLER_NAMESPACE

Reply:
@dhaiducek  you right I was confused with `KIND_NAME`

Reply:
In the normal  KIND_NAMESPACE use CONTROLLER_NAMESPACE. What is your suggestion? My brain pause right now to think better way

Reply:
Oh also the reason why I exclude `make kind-deploy-controller build-images;\`, in the hosted mode already the controller is deployed at https://github.com/stolostron/governance-policy-framework/blob/0903452ed5de2b5d99eba286f4a16ef1007ed064/.github/workflows/kind.yml#L110 so that is duplicate and useless time consuming

Reply:
I'm wondering about swapping these variables:

https://github.com/stolostron/config-policy-controller/blob/8155dac7e75bfd7ede26d0a56d7cd0f5091df48c/Makefile#L24-L29

Make `KIND_NAMESPACE` specify `open-cluster-management-agent-addon` so that the Framework can override it and set `CONTROLLER_NAMESPACE` to equal the value of `KIND_NAMESPACE`.

Reply:
so that case also need condition right?

Reply:
I just wonder why do we need 2 separate namespaces?

Reply:
My vote would be to:
1. Swap the variable values (The `CONTROLLER_NAMESPACE` variable is originally from the README, which is a name that makes more sense when deploying): https://github.com/stolostron/config-policy-controller/blob/8155dac7e75bfd7ede26d0a56d7cd0f5091df48c/README.md?plain=1#L144-L148
1.  Remove the `build-images` but leave `kind-deploy-controller` and figure out a different place for it to go OR remove `build-images` from the framework and make it consistent throughout our repos so that calling `kind-deploy-controller-dev` also calls `build-images`. I might lean toward the latter since when developing, I'd want to rebuild the image each time I redeploy the controller since I've probably changed the code and want to see the changes in my cluster, but I could go either way.

Reply:
but we don't need kind-deploy-controller for hosted mode. governance-policy-addon-controller deploy `deployment`

Reply:
I think we can delete `build-images` if `kind-deploy-controller ` is only used in framework test.  `build-images` actually duplicate. You can see this link
 https://github.com/stolostron/governance-policy-framework/blob/0903452ed5de2b5d99eba286f4a16ef1007ed064/.github/workflows/kind.yml#L160

Reply:
We still need `kind-deploy-controller` because if the deployment YAML changed in the PR, it's not picked up unless it's re-applied.

Okay, so you agree we remove `build-images` from the framework workflow and make sure it's called in each repo instead inside of the `kind-deploy-controller-dev` target? That sounds like a good plan to me.

Reply:
Oh, I see what you're saying. If the addon-controller is involved, then I might need to think more on what all is happening here...

Reply:
Thanks!

Reply:
oh should I remove `build-images` from the framework workflow I thought in controller.  Your suggestion is much better

Reply:
OH! I removed `build-images ` Thanks Dale